File: 22.linq-workshop-kata/java/src/test/java/linq_workshop/queries/Part1Test.java
package linq_workshop.queries;

import linq_workshop.queries.data.Person;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Predicate;

import static linq_workshop.queries.data.Persons.PEOPLE;
import static linq_workshop.queries.data.Persons.getPersonNamed;
import static org.assertj.core.api.Assertions.assertThat;

class Part1Test {

    @Test
    void getFirstNamesOfAllPeople() {
        List<String> firstNames = new ArrayList<>();

        assertThat(firstNames).containsExactlyInAnyOrder(
            "Mary", "Bob", "Ted", "Jake", "Barry", "Terry", "Harry", "John"
        );
    }

    @Test
    void getNamesOfMarySmithsPets() {
        var person = getPersonNamed("Mary Smith");

        List<String> names = new ArrayList<>();

        assertThat(names).containsExactly("Tabby");
    }

    @Test
    void getPeopleWithCats() {
        List<String> peopleWithCats = new ArrayList<>();

        assertThat(peopleWithCats).hasSize(2);
    }

    @Test
    void getPeopleWithoutCats() {
        List<String> peopleWithoutCats = new ArrayList<>();

        assertThat(peopleWithoutCats).hasSize(6);
    }

    @Test
    void doAnyPeopleHaveCats() {
        boolean doAnyPeopleHaveCats = false;

        assertThat(doAnyPeopleHaveCats).isTrue();
    }

    @Test
    void doAllPeopleHavePets() {
        Predicate<Person> predicate = p -> true;
        boolean result = PEOPLE.stream().allMatch(predicate);

        assertThat(result).isFalse();
    }

    @Test
    void howManyPeopleHaveCats() {
        long count = 0;

        assertThat(count).isEqualTo(2);
    }

    @Test
    void findMarySmith() {
        Person result = null;

        assertThat(result.firstName()).isEqualTo("Mary");
        assertThat(result.lastName()).isEqualTo("Smith");
    }

    @Test
    void getPeopleWithPets() {
        long petPeople = 0;

        assertThat(petPeople).isEqualTo(7);
    }
}

File: 22.linq-workshop-kata/ts/tests/02-queries/part1.test.ts
import { describe, expect, it } from "vitest";
import { people, getPersonNamed } from "../../src/02-queries/data/persons";
import { Person } from "../../src/02-queries/data/person";
import { sameMembers } from "../support/collections";

describe("Part1", () => {
  it("getFirstNamesOfAllPeople", () => {
    const firstNames: string[] = [];

    sameMembers(firstNames, ["Mary", "Bob", "Ted", "Jake", "Barry", "Terry", "Harry", "John"]);
  });

  it("getNamesOfMarySmithsPets", () => {
    const person = getPersonNamed("Mary Smith");

    const names: string[] = [];

    expect(names).toEqual(["Tabby"]);
  });

  it("getPeopleWithCats", () => {
    const peopleWithCats: string[] = [];

    expect(peopleWithCats).toHaveLength(2);
  });

  it("getPeopleWithoutCats", () => {
    const peopleWithoutCats: string[] = [];

    expect(peopleWithoutCats).toHaveLength(6);
  });

  it("doAnyPeopleHaveCats", () => {
    const doAnyPeopleHaveCats = false;

    expect(doAnyPeopleHaveCats).toBe(true);
  });

  it("doAllPeopleHavePets", () => {
    const predicate = (_person: Person): boolean => true;
    const result = people.every(predicate);

    expect(result).toBe(false);
  });

  it("howManyPeopleHaveCats", () => {
    const count = 0;

    expect(count).toBe(2);
  });

  it("findMarySmith", () => {
    const result: Person | undefined = undefined;

    expect(result!.firstName).toBe("Mary");
    expect(result!.lastName).toBe("Smith");
  });

  it("getPeopleWithPets", () => {
    const petPeople = 0;

    expect(petPeople).toBe(7);
  });
});
